### PR TITLE
feat(tofs): add TOFS support

### PR DESCRIPTION
**Pull request by n-rodriguez**

### CODE REVIEW

**Comment by n-rodriguez** [comment]
> Can we keep the filename the same for the time being? Meaning:

I thought the convention was to use `<filename>.jinja.tmpl`

**Comment by myii** [comment]
> I thought the convention was to use `<filename>.jinja.tmpl`

@n-rodriguez Actually, we haven't quite decided a convention yet for this.  When we're refactoring to TOFS, I prefer to keep the template filename identical, to minimise confusion when reviewing and for end-users.  If there is a pressing need to change the template filename, that can always be covered in a subsequent PR.

File: pillar.debian.example [hunk 1]
```diff
@@ -2,6 +2,32 @@
 # vim: ft=yaml
 ---
 syslog_ng:
+  # tofs:
```
Comment (myii) [inline pillar.debian.example]: Can we have this block added to the end of `pillar.example` instead?

File: pillar.debian.example [hunk 2]
```diff
@@ -2,6 +2,32 @@
...
+  #   This is unnecessary in most cases; there are sensible defaults.
+  #   path_prefix: template_alt
+  #   dirs:
+  #     files: files_alt
+  #     default: default_alt
+  #   The entries under `source_files` are prepended to the default source files
+  #   given for the state
+  #   source_files:
+  #     syslog_ng/config/install:
+  #       - 'example_alt.tmpl'
```
Comment (myii) [inline pillar.debian.example]: Just the one entry here (i.e. remove the line underneath).  We tend to use a modified version of the actual template name, so something like:

```suggestion
  #       - 'alt_syslog-ng.conf'
```

**Comment by myii** [comment]
@n-rodriguez Just checked and found that we discussed the `source` in the original TOFS PR (#14):

* https://github.com/saltstack-formulas/syslog-ng-formula/pull/14#discussion_r288345613

**Comment by myii** [comment]
@n-rodriguez I added another commit since there was a `yamllint` error:

* https://travis-ci.com/saltstack-formulas/syslog-ng-formula/builds/129754190#L217-L219

If the tests all pass, I'll merge this.

**Comment by myii** [comment]
@n-rodriguez All working, thanks -- merged.

**Pull request merged**